lib/mayaUsd/ufe/Utils.cpp

```diff
@@ -1139,5 +1147,107 @@ void ReplicateExtrasToUSD::finalize(
 #endif
 }
 
+#ifdef HAS_ORPHANED_NODES_MANAGER
+
+static Ufe::BBox3d transformBBox(const PXR_NS::GfMatrix4d& matrix, const Ufe::BBox3d& bbox)
+{
+    Ufe::BBox3d transformed(bbox);
+
+    transformed.min = toUfe(matrix.Transform(toUsd(bbox.min)));
+    transformed.max = toUfe(matrix.Transform(toUsd(bbox.max)));
+
+    return transformed;
+}
+
+static Ufe::BBox3d transformBBox(const Ufe::Matrix4d& matrix, const Ufe::BBox3d& bbox)
+{
+    return transformBBox(toUsd(matrix), bbox);
+}
+
+static Ufe::BBox3d transformBBox(Ufe::SceneItem::Ptr& item, const Ufe::BBox3d& bbox)
+{
+    Ufe::Transform3d::Ptr t3d = Ufe::Transform3d::transform3d(item);
+    if (!t3d)
+        return bbox;
+
+    return transformBBox(t3d->matrix(), bbox);
+}
+
+#ifdef MAYA_OBJECT3D_HAS_BOUNDING_BOX
+
+static Ufe::BBox3d getTransformedBBox(const Ufe::Path& path)
+{
+    Ufe::SceneItem::Ptr item = Ufe::Hierarchy::createItem(path);
+    if (!item)
+        return {};
+
+    const auto&               runtimeMgr = Ufe::RunTimeMgr::instance();
+    Ufe::Object3dHandler::Ptr obj3dHandler = runtimeMgr.object3dHandler(item->runTimeId());
+    if (!obj3dHandler)
+        return {};
+
+    Ufe::Object3d::Ptr o3d = obj3dHandler->object3d(item);
+    if (!o3d)
+        return {};
+
+    return transformBBox(item, o3d->boundingBox());
+}
+
+#else
+
+static Ufe::BBox3d getTransformedBBox(const MDagPath& path)
+{
+    MFnDagNode node(path);
+
+    MBoundingBox mayaBBox = node.boundingBox();
+    return Ufe::BBox3d(
+        Ufe::Vector3d(mayaBBox.min().x, mayaBBox.min().y, mayaBBox.min().z),
+        Ufe::Vector3d(mayaBBox.max().x, mayaBBox.max().y, mayaBBox.max().z));
+}
+
+#endif /* MAYA_OBJECT3D_HAS_BOUNDING_BOX */
+
+#endif /* HAS_ORPHANED_NODES_MANAGER */
+
+Ufe::BBox3d getPulledPrimsBoundingBox(const Ufe::Path& path)
+{
+    Ufe::BBox3d ufeBBox;
+
+#ifdef HAS_ORPHANED_NODES_MANAGER
+    const auto& updaterMgr = PXR_NS::PrimUpdaterManager::getInstance();
+    PXR_NS::PrimUpdaterManager::PulledPrimPaths pulledPaths = updaterMgr.getPulledPrimPaths();
+    for (const auto& paths : pulledPaths) {
+        const Ufe::Path& pulledPath = paths.first;
+
+        if (pulledPath == path)
+            continue;
+
+        if (!pulledPath.startsWith(path))
+            continue;
+
+            // Note: Maya implementation of the Object3d UFE interface does not
+            //       implement the boundingBox() function. So we ask the DAG instead.
+#ifdef MAYA_OBJECT3D_HAS_BOUNDING_BOX
+        Ufe::BBox3d pulledBBox = getTransformedBBox(pulledPath);
+#else
```

Comment:
Why did you leave this code (and above)? It is not used and never will be. We won't ever implement the boundingBox in Maya Object3d interface. Just leave the comment.